operators/cmd/crownlabs-delete-labinstance/deleteCron.py

```diff
@@ -0,0 +1,190 @@
+        """
+        api_instance = client.CustomObjectsApi()
+        try:
+            template = api_instance.get_namespaced_custom_object(group=group, version=version,
+                                                                 namespace=template_ns, plural=plural_template,
+                                                                 name=template_name)
+        except ApiException as e:
+            logger.debug(f"Exception when calling CustomObjectsApi->get_namespaced_custom_object: '{e}'")
+            return None
+        delete_after = template.get("spec").get("deleteAfter")
+        time = self.convert_to_time(delete_after)
+        return time
+
+    def _delete_instance_expired(self, instances):
+        """
+        Deletes from current list of instances the exiperd ones.
+        :returns: None
+        :instances: current instance list in crowlabs
+        """
+        api_instance = client.CustomObjectsApi()
+        for instance in instances.get("items"):
+            # get instance name
+            name = instance.get("metadata").get("name")
+            # get instance creation timestamp
+            creation_timestamp = instance.get("metadata").get("creationTimestamp")
+            # get instance
+            namespace = instance.get("metadata").get("namespace")
+            # retrieve template name
+            template_name = instance.get("spec").get("template.crownlabs.polito.it/TemplateRef").get("name")
+            # retrieve template namespace
+            template_ns = instance.get("spec").get("template.crownlabs.polito.it/TemplateRef").get("namespace")
+            if template_ns is None:
+                template_ns = 'default'
+            # retrieve instance life span
+            time = self._get_life_span(template_name, template_ns)
+            if time is None:
+                logger.debug(f"template: '{template_name}' -in Namespace: '{template_ns}'\
+                             has a wrong delete_after field format")
+            else:
+                # verify labinstance expiration status
+                if(self.instance_is_expired(time, creation_timestamp)):
+                    try:
+                        # delete expired labinstance
+                        # check run mode
+                        if self.dry_run:
+                            api_instance.delete_namespaced_custom_object(group=group, version=version,
+                                                                         namespace=namespace, plural=plural_instance,
+                                                                         name=name, dry_run='All')
+                        else:
+                            api_instance.delete_namespaced_custom_object(group=group, version=version,
+                                                                         namespace=namespace, plural=plural_instance,
+                                                                         name=name)
```

Comment:
This code duplication should be avoided. You could set the dry_run variable to `None` or `'All'` in the constructor and directly use the variable